### PR TITLE
(Wallet) Enable Filecoin support flag

### PR DESCRIPTION
**Pull request by simoarpe**
## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
* Navigate to the Brave Wallet and create a new Filecoin testnet account
* Add Filecoins using a faucet service
* Send a Filecoin amount to a valid Filecoin testnet address
* Verify the unapproved transaction is created
* Approve the transaction
* Navigate to the Accounts section
* Tap on the Filecoin testnet account
* Lookup for the transaction and tap on it
* Verify the link to the blockchain explorer opens correctly

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by Pavneet-Sing — COMMENTED** [review]
File coin test net send has issue with network, asset, and balance. Desktop has no asset for filecoin test net send, the filecoin test case works?

[file-coin-testnet.webm](https://github.com/brave/brave-core/assets/11755381/d78b94d6-ebf1-49b7-86ea-2f4ef3e7b044)

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransactionFees.java [hunk 1]
```diff
@@ -130,20 +150,43 @@ public static ParsedTransactionFees parseTransactionFees(TransactionInfo txInfo,
...
-                                ? Utils.multiplyHexBN(maxFeePerGas, gasLimit)
-                                : Utils.multiplyHexBN(gasPrice, gasLimit),
-                        networkDecimals);
+        double gasFee;
+        if (isSolTransaction) {
+            gasFee = solFeeEstimatesFee != 0
+                    ? Utils.fromWei(Long.toString(solFeeEstimatesFee), networkDecimals)
+                    : 0.0d;
+        } else {
+            // In common for Eth and Fil as calculations are the same.
```
[the remark this answers is not in the document]
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransactionFees.java:184]: Yeah, not very clear. I'm rephrasing this sentence.

**Comment by simoarpe** [comment]
> File coin test net send has issue with network, asset, and balance. Desktop has no asset for filecoin test net send, the filecoin test case works?

Thanks, on it!

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransactionFees.java [hunk 2]
```diff
@@ -130,20 +150,43 @@ public static ParsedTransactionFees parseTransactionFees(TransactionInfo txInfo,
...
 
+    @NonNull
+    private static String calculateFilGasPrice(@NonNull final String maxFeePerGasDecimal,
+            @NonNull final String maxPriorityFeePerGasDecimal) {
+        if (maxFeePerGasDecimal.isEmpty() || maxPriorityFeePerGasDecimal.isEmpty()) {
+            return "";
+        } else {
+            String result;
+            try {
+                BigInteger minued = new BigInteger(maxFeePerGasDecimal);
```
Comment (supermassive) [inline android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransactionFees.java:161]: minuend

Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransactionFees.java]: Fixed, thanks!

File: android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java [hunk 3]
```diff
@@ -287,29 +287,10 @@ private void addAccountInternal(@CoinType.EnumType int coinType,
                 result -> { handleAddAccountResult(result, callback); });
     }
 
-    private static @KeyringId.EnumType int getTargetKeyringId(
```
Comment (simoarpe) [inline android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java:290]: Method removed as duplicate of `AssetUtils.getKeyring`.

**Pull request merged**